backend/src/main/java/de/jmpsoftware/backend/service/ArtNetThread.java

```diff
@@ -0,0 +1,123 @@
+
+        LOG.trace("ArtNetClient is created");
+    }
+
+    @SneakyThrows
+    @Override
+    public void run() {
+        LOG.trace("ArtNetClient Thread goes run");
+        this.artnet.start();
+        sendDMXTimeStamp = ZonedDateTime.now().toInstant().toEpochMilli();
+
+        while (true) {
+            while (pipe.available() > 0) {
+                if (checkPipeReadDoneCorrect(pipe.read(readByte, 0, 4))) {
+
+                    pipeEntry.setValuesFromBytePack(readByte);
+
+                    LOG.info("ThreadRead" + pipeEntry);
+
+                    if(isPipeEntryCommandDMX(pipeEntry)) {
+
+                        if (pipeEntry.getValuesValid())
+                            dmxData[pipeEntry.getUniverse()][pipeEntry.getChannel()] = (byte) pipeEntry.getValueDMX();
+
+                        if (!pipeEntry.isCommand(ArtNetService.COMMANDO_DMX_NO_SEND))
+                            broadcastDmx();
+                    } else {
+                        if (pipeEntry.isCommand(ArtNetService.COMMAND_NO_STOP))
+                            this.artnet.stop();
+                        else if (pipeEntry.isCommand(ArtNetService.COMMAND_NO_START))
+                            this.artnet.start();
+                    }
+                }
+            }
+            if (this.artnet.isRunning()) {
+                if (isMaxTimeBetweenLastDmxSendBigger(ARTNET_FORCE_BROADCAST_TIME)) {
+                    broadcastDmx();
+                } else {
+                    sleepThreadMillis(ACTIVE_ARTNET_DOZY_TIME);
+                }
+            } else {
+                sleepThreadMillis(SLEEPING_ARTNET_DOZY_TIME);
+            }
+        }
+    }
+
+    private void sleepThreadMillis(long millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException e) {
+            LOG.warn("Do not Panic! But Sleep did not work.");
+        }
+    }
+
+    private boolean isMaxTimeBetweenLastDmxSendBigger(long millis) {
+        return ((ZonedDateTime.now().toInstant().toEpochMilli()) - sendDMXTimeStamp) > millis;
+    }
+
+    boolean isPipeEntryCommandDMX(PipeEntry pipeEntry) {
+        return ((pipeEntry.getCommand() >= ArtNetService.COMMAND_DMX) && (pipeEntry.getCommand() <= ArtNetService.COMMANDO_DMX_NO_SEND));
+
+    }
+
+    boolean checkPipeReadDoneCorrect(int pipeReadReturn) throws IOException {
+        if (pipeReadReturn == -1) {
+            return false;
+        }
+        if (pipeReadReturn != 4) {
+            LOG.warn("Thread Pipe did not Read correct Data");
+            LOG.info("Thread Pipe Data may be Corrupt! Cleanup All Data");
+            byte[] buffer = new byte[100];
+            while (pipe.available() > 0)
+            {
+                int trashIt = pipe.read(buffer,0,100);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    public void broadcastDmx() {
+        // UNIVERSE Startet mit 1 !!!!!!!!!
+        // ToDO Active Universe Handling
```

Comment:
Track your todos somewhere else, it is too easy to forget to delete these comments and reach a point where no one knows if this is still open or not